### ISSUE TITLE
base config is not documented

### ISSUE THREAD

**Issue by ssbarnea** [ansible/molecule#1841]
Current base configuration is fully undocumented. The page that is most likely to describe it contain nothing https://molecule.readthedocs.io/en/latest/configuration.html and neither other page of the documentation.

We not only need to document this but also make molecule load its base configuration automatically by discovering it from the code repository.

For local development is ok to keep configuration in a `~/.config` but any colaborative project would expect to have a git hosted base config file that would be found and used when molecule is run.

**Comment by decentral1se** [comment]
Asked for in https://github.com/ansible/molecule/issues/1423#issuecomment-474963763 as well :+1:

**Comment by tace** [comment]
Good there is already ticket for this!
+1 for the documentation.
Trying to try/error how it works. 

Please document also --env-file option.

**Comment by decentral1se** [comment]
Right, we have auto-loading already:
  * https://github.com/ansible/molecule/blob/master/molecule/config.py#L312
  * https://github.com/ansible/molecule/blob/master/molecule/config.py#L514

I'd like to suggest then to close this off that we provide new default paths:
  * `${MOLECULE_PROJECT_DIRECTORY}/.molecule/molecule.yml` # base-config
  * `${MOLECULE_PROJECT_DIRECTORY}/.molecule/env.yml`  # env-file

And keep loading from the existing defaults to not break anyones use case.

**Comment by WithoutAnAce** [comment]
Is there a default `env.yml` path already, or is this a proposed change?

**Comment by dzintars** [comment]
Yeah, `molecule --base-config .config/molecule/config.yml converge -s dev -- --limit workstations --tags nushell --list-tasks` is a little bit verbose. I would like to drop `base-config` and `scenario` flags.

**Comment by dzintars** [comment]
Similar issue https://github.com/ansible-community/molecule/issues/1664
